airflow/utils/python_virtualenv.py

```diff
@@ -118,6 +118,7 @@ def write_python_script(
         to a native Python object
     """
     template_loader = jinja2.FileSystemLoader(searchpath=os.path.dirname(__file__))
+    template_env: Union[jinja2.nativetypes.NativeEnvironment, jinja2.Environment]
```

Comment:
```suggestion
    template_env: jinja2.Environment
```

Because `NativeEnvironment` is a subclass of `Environment`